compiler-rt/lib/fuzzer/FuzzerUtilFuchsia.cpp

```diff
@@ -607,7 +607,11 @@ size_t PageSize() {
 }
 
 void SetThreadName(std::thread &thread, const std::string &name) {
-  // TODO ?
+  if (name.size() > 31) {
+    name.resize(31);
+  }
+  (void)zx_object_set_property(thread.native_handle(), ZX_PROP_NAME,
```

Comment:
`zx_object_set_property` can fail so rather than ignoring the return value, we should at least log it. It'd be even better if `SetThreadName` could return a status but that would require further refactoring.

Reply:
Yes refactoring might be a nice next step what you say makes sense but now it s a bit inconsistent with Linux version for example.